src/renderer/dx/DxRenderer.cpp

```diff
@@ -147,39 +149,38 @@ DxEngine::~DxEngine()
                               // D3D11_CREATE_DEVICE_DEBUG |
                               D3D11_CREATE_DEVICE_SINGLETHREADED;
 
-    D3D_FEATURE_LEVEL FeatureLevels[] = {
-        D3D_FEATURE_LEVEL_11_1,
-        D3D_FEATURE_LEVEL_11_0,
-        D3D_FEATURE_LEVEL_10_1,
-        D3D_FEATURE_LEVEL_10_0,
-        D3D_FEATURE_LEVEL_9_1,
-    };
+    std::array<D3D_FEATURE_LEVEL, 5> FeatureLevels;
+    FeatureLevels.at(0) = D3D_FEATURE_LEVEL_11_1;
+    FeatureLevels.at(1) = D3D_FEATURE_LEVEL_11_0;
+    FeatureLevels.at(2) = D3D_FEATURE_LEVEL_10_1;
+    FeatureLevels.at(3) = D3D_FEATURE_LEVEL_10_0;
+    FeatureLevels.at(4) = D3D_FEATURE_LEVEL_9_1;
 
     // Trying hardware first for maximum performance, then trying WARP (software) renderer second
     // in case we're running inside a downlevel VM where hardware passthrough isn't enabled like
     // for Windows 7 in a VM.
-    const auto hardwareResult = D3D11CreateDevice(NULL,
+    const auto hardwareResult = D3D11CreateDevice(nullptr,
                                                   D3D_DRIVER_TYPE_HARDWARE,
-                                                  NULL,
+                                                  nullptr,
                                                   DeviceFlags,
-                                                  FeatureLevels,
-                                                  ARRAYSIZE(FeatureLevels),
+                                                  FeatureLevels.data(),
+                                                  gsl::narrow<UINT>(FeatureLevels.size()),

```

Comment:
this can be a narrow_cast; given that it's a static array with the size set above, we don't _need_ to fail if somebody hardcodes more than 4 billion feature levels.

Reply:
done